ui/components/multichain/pages/review-permissions-page/review-permissions-page.tsx

```
@@ -0,0 +1,298 @@
+  const { openMetaMaskTabs }: { openMetaMaskTabs: Record<string, boolean> } =
+    useSelector(
+      (state: { appState: { openMetaMaskTabs: Record<string, boolean> } }) =>
+        state.appState,
+    );
+
+  const { id }: { id: string } = useSelector(
+    (state: { activeTab: { id: string } }) => state.activeTab,
+  );
+
+  const subjectMetadata: { [key: string]: any } = useSelector(
+    getConnectedSitesList,
+  );
+  const connectedSubjectsMetadata = subjectMetadata[activeTabOrigin];
+
+  const connectedNetworks = useSelector((state) =>
+    getPermittedChainsForSelectedTab(state, activeTabOrigin),
+  ) as string[];
+
+  const permittedAccountsByOrigin = useSelector(
+    getPermittedAccountsByOrigin,
+  ) as PermittedAccountsByOrigin;
+  const networksList = useSelector(getNonTestNetworks);
+  const testNetworks = useSelector(getTestNetworks);
+  const combinedNetworks = [...networksList, ...testNetworks];
+
+  const connectedAccounts = useSelector((state) =>
+    getOrderedConnectedAccountsForConnectedDapp(state, activeTabOrigin),
+  ) as AccountType[];
+
+  const subjects = useSelector(getPermissionSubjects);
+  const grantedNetworks = combinedNetworks.filter(
+    (net: { chainId: any }) => connectedNetworks.indexOf(net.chainId) !== -1,
+  );
+
+  const hostName = getURLHost(securedOrigin);
+  const currentTabHasNoAccounts =
+    !permittedAccountsByOrigin[activeTabOrigin]?.length;
+
+  let tabToConnect: { origin: string | null } = { origin: null };
+  if (activeTabOrigin && currentTabHasNoAccounts && !openMetaMaskTabs[id]) {
+    tabToConnect = {
+      origin: activeTabOrigin,
+    };
+  }
+
+  const requestAccountsAndChainPermissions = async () => {
+    if (tabToConnect.origin) {
+      // Ensure origin is not null
+      const requestId = await dispatch(
+        requestAccountsAndChainPermissionsWithId(tabToConnect.origin),
+      );
+      history.push(`${CONNECT_ROUTE}/${requestId}`);
+    } else {
+      console.error('Tab origin is null.');
```

Should we throw an error here?  I don't know that `console.error` is helpful.

---

This shouldn't throw an error. Because it's just a simple route. Remove the else condition